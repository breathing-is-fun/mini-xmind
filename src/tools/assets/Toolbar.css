.Toolbar ul {
  list-style: none;
  position: fixed;
  left: 5px;
  top: 40%;
  z-index: 3;
}

.Toolbar ul li {
  cursor: pointer;
  margin-bottom: 10px;
}
