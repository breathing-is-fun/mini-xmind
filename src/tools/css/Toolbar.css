.Toolbar ul {
  list-style: none;
  position: fixed;
  left: 5px;
  top: 40%;
  z-index: 3;
}

.Toolbar ul li {
  cursor: pointer;
  margin-bottom: 10px;
}

.Toolbar .icon-minus {
  -webkit-mask-image: url(../svg/minus.svg);
  mask-image: url(../svg/minus.svg);
  width: 20px;
  height: 20px;
  display: inline-block;
  background: gray;
}

.Toolbar .icon-border {
  -webkit-mask-image: url(../svg/border.svg);
  mask-image: url(../svg/border.svg);
  width: 20px;
  height: 20px;
  display: inline-block;
  background: gray;
}

.Toolbar .icon-tag {
  -webkit-mask-image: url(../svg/tag.svg);
  mask-image: url(../svg/tag.svg);
  width: 20px;
  height: 20px;
  display: inline-block;
  background: gray;
}

.Toolbar .icon-branches {
  -webkit-mask-image: url(../svg/branches.svg);
  mask-image: url(../svg/branches.svg);
  width: 20px;
  height: 20px;
  display: inline-block;
  background: gray;
}
