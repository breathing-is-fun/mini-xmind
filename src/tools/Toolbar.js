import React, { Component } from 'react';

import { Icon } from 'antd';
import { tools } from '../options/tools';

import './assets/Toolbar.css';

export default class Toolbar extends Component {
  static propTypes = {};

  static defaultProps = {};

  onDragStart = (e, item) => {
    e.dataTransfer.effectAllowed = 'copy';

    e.dataTransfer.setData('dragItem', JSON.stringify(item));
  };

  render = () => {
    return (
      <div className="Toolbar">
        <ul>
          {tools.map((item, i) => {
            const { key, value } = item;
            return (
              <li
                draggable
                onDragStart={e => this.onDragStart(e, item)}
                key={`tool-${i}`}
              >
                <Icon
                  type={key}
                  // onClick={() => this.handleIconClick(value)}
                  style={{
                    fontSize: 20,
                  }}
                />
              </li>
            );
          })}
        </ul>
      </div>
    );
  };
}
