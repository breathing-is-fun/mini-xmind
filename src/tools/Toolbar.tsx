import React from 'react';
import './css/Toolbar.css';

export type DraggableItemProp = 'block' | 'tag';

const draggableItem: {
  type: DraggableItemProp;
  item: React.ReactElement;
  key: string;
}[] = [
  { type: 'block', item: <div className="icon-border" />, key: 'toolbar-block' },
  { type: 'tag', item: <div className="icon-tag" />, key: 'toolbar-tag' },
];

const onDragStart = (e: React.DragEvent<HTMLLIElement>, item: { type: DraggableItemProp }) => {
  e.dataTransfer.effectAllowed = 'copy';
  e.dataTransfer.setData('dragItem', JSON.stringify(item));
};

const Toolbar = () => (
  <div className="Toolbar">
    <ul>
      <li>
        <div className="icon-minus" />
      </li>
      <li>
        <div className="icon-branches" />
      </li>
      {draggableItem.map(({ type, item, key }) => (
        <li draggable onDragStart={e => onDragStart(e, { type })} key={key}>
          {item}
        </li>
      ))}
    </ul>
  </div>
);

export default Toolbar;
