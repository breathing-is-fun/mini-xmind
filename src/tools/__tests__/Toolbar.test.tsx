import React from 'react';
import { mount } from 'enzyme';
import Toolbar from '../Toolbar';

describe('Toolbar', () => {
  it('toolbar render correctly', () => {
    const wrapper = mount(<Toolbar />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.icon-minus')).toBeTruthy();
    expect(wrapper.find('.icon-branches')).toBeTruthy();
    expect(wrapper.find('.icon-border')).toBeTruthy();
    expect(wrapper.find('.icon-tag')).toBeTruthy();
  });

  it('onDragStart', () => {
    const wrapper = mount(<Toolbar />);
    const onChange = jest.fn();
    const setData = function() {
      onChange();
    };
    const event = { dataTransfer: { effectAllowed: '', setData } };
    (wrapper
      .find('li')
      .at(2)
      .props() as any).onDragStart(event, '{ test: 1 }');
    expect(onChange).toBeCalled();
    expect(event.dataTransfer.effectAllowed).toBe('copy');
  });
});
