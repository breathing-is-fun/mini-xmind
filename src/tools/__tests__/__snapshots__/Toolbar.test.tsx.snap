// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar toolbar render correctly 1`] = `
<Toolbar>
  <div
    className="Toolbar"
  >
    <ul>
      <li>
        <div
          className="icon-minus"
        />
      </li>
      <li>
        <div
          className="icon-branches"
        />
      </li>
      <li
        draggable={true}
        key="toolbar-block"
        onDragStart={[Function]}
      >
        <div
          className="icon-border"
        />
      </li>
      <li
        draggable={true}
        key="toolbar-tag"
        onDragStart={[Function]}
      >
        <div
          className="icon-tag"
        />
      </li>
    </ul>
  </div>
</Toolbar>
`;
